src/iter.rs

```diff
@@ -221,9 +224,9 @@ where
                 Some(c)
             }
             None => self.iter.next().map(|b| {
-                let (high, low) = byte_to_hex_chars(b);
-                self.low = Some(low);
-                high
+                let [high, low] = self.table.byte_to_hex(b);
+                self.low = Some(low as char);
```

Comment:
`char::from(low)` would be more clear here because makes it explicit that no data is being lost because it is infallible. As it is, the reader has to look up what type `low` is and then look up (or know) what casting that type to a `char` does.

Reply:
Some below for `high`.